packages/modules/viewer-react/src/components/BaseViewer.tsx

```diff
@@ -29,11 +30,13 @@ export const BaseViewer = ({
   });
 
   const accessToken = useAccessToken();
+  const hasBackstageItems = loaderProps.backstageItems ? (loaderProps.backstageItems.length > 0 || UiItemsManager.getBackstageItems().length > 0) : UiItemsManager.getBackstageItems().length > 0;
+  const cornerButton = hasBackstageItems ? <BackstageAppButton /> : undefined
   return (
     <ErrorBoundary>
       {loaderProps.filePath || accessToken ? (
         viewerInitialized ? (
-          <IModelLoader {...loaderProps} />
+          <IModelLoader defaultUiConfig={{cornerButton}} {...loaderProps}  />
```

Comment:
you def dont need to make changes in this file;
youre duplicating the work of hasBackstageItems here and in IModelViewer
take a look at what the type of loaderProps is.